src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```
@@ -185,6 +185,10 @@ private function parseDefinition($id, $service, $file)
                 $parameters[$name] = SimpleXMLElement::phpize($value);
             }
 
+            if ('' === trim($tag['name'])) {
+                throw new \InvalidArgumentException(sprintf('The tag name for service "%s" must not be an empty string in %s.', $id, $file));
```

I think this should throw the components InvalidArgumentException